src/libraries/Microsoft.PowerFx.Core/Public/Types/UserProvidedTypeDefinitions/UserDefinedRecordType.cs

```diff
@@ -0,0 +1,50 @@
+﻿// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using Microsoft.PowerFx.Core.Utils;
+using Microsoft.PowerFx.Types;
+
+namespace Microsoft.PowerFx.Core
+{
+    internal class UserDefinedRecordType : RecordType
+    {
+        private readonly FormulaTypeSchema _backingSchema;
+        private readonly DefinedTypeSymbolTable _symbolTable;
+
+        public override IEnumerable<string> FieldNames => _backingSchema.Fields?.Keys ?? Enumerable.Empty<string>();
+
+        public UserDefinedRecordType(FormulaTypeSchema backingSchema, DefinedTypeSymbolTable definedTypes)
+        {
+            _backingSchema = backingSchema;
+            _symbolTable = definedTypes;
+        }
+
+        public override bool TryGetFieldType(string name, out FormulaType type)
+        {
+            if (_backingSchema.Fields.TryGetValue(name, out var fieldType))
+            {
+                type = fieldType.ToFormulaType(_symbolTable);
+                return true;
+            }
+            
+            type = FormulaType.Blank;
```

Comment:
>FormulaType.Blank; [](http://example.com/codeflow?start=19&length=18)

nit: normal convention is false means the out  parameter is null.

Reply:
Not in the FormulaType hierarchy, it seems. AggregateType.TryGetFieldType also sets the out param to Blank